src/core/Dropdown/Dropdown/Dropdown.tsx

```diff
@@ -241,37 +195,85 @@ export class Dropdown extends Component<DropdownProps> {
         id={id}
         {...passProps}
       >
-        <HtmlLabel
+        <LabelText
           id={labelId}
-          {...labelProps}
-          className={dropdownClassNames.label}
+          labelMode={labelMode}
+          as="label"
+          optionalText={optionalText}
         >
-          {labelMode === 'hidden' ? (
-            <VisuallyHidden>{labelText}</VisuallyHidden>
-          ) : (
-            <Paragraph {...labelTextProps}>{labelText}</Paragraph>
-          )}
-        </HtmlLabel>
+          {labelText}
+        </LabelText>
         <ListboxInput {...listboxInputProps}>
-          <ListboxButton {...passDropdownButtonProps}>
+          <ListboxButton
+            {...dropdownButtonProps}
+            id={buttonId}
+            className={classnames(
+              dropdownClassNames.button,
+              dropdownButtonProps.className,
+              {
+                [dropdownClassNames.disabled]: !!disabled,
+              },
+            )}
+            {...buttonAriaLabelledByOverride}
+          >
             {listboxDisplayValue}
           </ListboxButton>
           {!!ListboxPopoverComponentReplace ? (
             <ListboxPopoverComponentReplace {...passDropdownPopoverProps}>
-              {this.dropdownItems(children, passDropdownItemProps)}
+              {children}
             </ListboxPopoverComponentReplace>
           ) : (
             <ListboxPopover
               position={positionMatchWidth}
               {...passDropdownPopoverProps}
             >
-              <ListboxList>
-                {this.dropdownItems(children, passDropdownItemProps)}
-              </ListboxList>
+              <ListboxList>{children}</ListboxList>
             </ListboxPopover>
           )}
         </ListboxInput>
       </HtmlSpan>
     );
   }
 }
+
+const StyledDropdown = styled(
+  ({
+    tokens,
+    id: propId,
+    ...passProps
+  }: DropdownProps & InternalTokensProp) => (
+    <AutoId id={propId}>
+      {(id) => <BaseDropdown id={id} {...passProps} />}
+    </AutoId>
+  ),
+)`
+  ${(props) => baseStyles(props)}
+`;
+
+interface NewListboxPopoverProps extends BaseListboxPopoverProps, TokensProp {}
+
+const StyledListboxPopover = styled(
+  ({ tokens, children, ...passProps }: NewListboxPopoverProps) => (
```

Comment:
Maybe we do not need the `NewListboxPopoverProps` as it does not introduce anything.

Instead of:
```
const StyledListboxPopover = styled(
    ({ tokens, children, ...passProps }: NewListboxPopoverProps) => (
```

We could use it without it, like this:
```
const StyledListboxPopover = styled(
    ({ tokens, children, ...passProps }: BaseListboxPopoverProps & TokensProp) => (
```

Reply:
Good point. I think that was from the old implementation. Changed it (not pushed yet).

Reply:
This could probably be dropped altogether? And should actually work with Partial<ListboxPopoverProps> & TokensProp or something similar.

Reply:
It's now gone.